include/galsim/LRUCache.h

```diff
@@ -22,11 +22,66 @@
 
 #include <list>
 #include <map>
-#include <boost/tuple/tuple.hpp>
-#include <boost/tuple/tuple_comparison.hpp>  // Need this for t1 < t2
 
 namespace galsim {
 
+    // A very simple tuple class that just does what we need for the LRUCache.
+    // It can hold up to a maximum of 4 parameters.
+    template <typename T1, typename T2=int, typename T3=int, typename T4=int>
+    class Tuple
+    {
+    public:
+        T1 first;
+        T2 second;
+        T3 third;
+        T4 fourth;
+
+        Tuple(const T1& a) : first(a), second(0), third(0), fourth(0) {}
+        Tuple(const T1& a, const T2& b) : first(a), second(b), third(0), fourth(0) {}
+        Tuple(const T1& a, const T2& b, const T3& c) : first(a), second(b), third(c), fourth(0) {}
+        Tuple(const T1& a, const T2& b, const T3& c, const T4& d) :
+            first(a), second(b), third(c), fourth(d) {}
+
+        Tuple(const Tuple& rhs) :
+            first(rhs.first), second(rhs.second), third(rhs.third), fourth(rhs.fourth) {}
+
+        Tuple& operator=(const Tuple& rhs)
+        {
+            if (&rhs != this) {
+                first = rhs.first;
+                second = rhs.second;
+                third = rhs.third;
+                fourth = rhs.fourth;
+            }
+            return *this;
+        }
+
+        bool operator<(const Tuple& rhs) const
+        {
+            return (
```

Comment:
It's the unrolled version of the [boost implementation](http://www.boost.org/doc/libs/1_62_0/boost/tuple/tuple_comparison.hpp), hard coded for 4 template arguments.  Hard to say whether boost's template magic makes the code easier or harder to follow.  The lt function is shorter, but you need to understand how the head and tail things work, which is slightly tricky.